crates/bevy_asset/src/io/gated.rs

```diff
@@ -36,7 +33,7 @@ impl GateOpener {
     pub fn open<P: AsRef<Path>>(&self, path: P) {
         let mut gates = self.gates.write();
         let gates = gates
-            .entry(path.as_ref().to_path_buf())
+            .entry(path.as_ref().into())
```

Comment:
```suggestion
            .entry_ref(path.as_ref())
```
You might be able to elide the the allocation entirely by using `entry_ref`.